app/scripts/directives/blocks.js

```diff
@@ -234,3 +246,104 @@ angular
       },
     };
   });
+
+angular
+  .module('confRegistrationWebApp')
+  .directive('opportunitiesQuestion', function () {
+    return {
+      templateUrl: opportunitiesQuestionTemplate,
+      restrict: 'E',
+      controller: function ($scope) {
+        $scope.block.content.staticChoices = [
+          'Yes, via email',
+          'Yes, via text',
+          'Yes, via email & text',
+          'No',
+        ];
+      },
+    };
+  });
+
+angular
+  .module('confRegistrationWebApp')
+  .directive('graduationDateQuestion', function () {
+    return {
+      templateUrl: graduationDateQuestionTemplate,
+      restrict: 'E',
+    };
+  });
+
+angular
+  .module('confRegistrationWebApp')
+  .directive('ethnicityQuestion', function () {
+    return {
+      templateUrl: ethnicityQuestionTemplate,
+      restrict: 'E',
+      controller: function ($scope, $timeout) {
+        $scope.otherSentinel = '__other__';
+
+        $scope.block.content.staticChoices = [
+          'American Indian/Native Alaskan',
+          'Arab/Middle Eastern',
+          'Asian/Asian American',
+          'Black/African American',
+          'Hispanic/Latino',
+          'Multi-racial/Multi-ethnic',
+          'Pacific Islander/Native Hawaiian',
+          'White/European American',
+        ];
+
+        $scope.selectOtherAnswer = () => {
+          if ($scope.otherAnswer) {
+            $scope.selectedAnswer = $scope.otherSentinel;
+          }
+        };
+
+        $scope.clearOther = () => {
+          if ($scope.selectedAnswer != $scope.otherSentinel)
+            $scope.otherAnswer = '';
+        };
+
+        const answerValue = $scope.answer ? $scope.answer.value : '';
+        if (
+          answerValue &&
+          !_.includes($scope.block.content.staticChoices, answerValue)
+        ) {
```

Comment:
We need to make an other choice, which will make this IF statement easier to understand.

Reply:
You will be able to make the function `selectOtherAnswer` simpler as well.